jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppContext.java

```diff
@@ -548,14 +548,14 @@ protected void doStart() throws Exception
             if (isLogUrlOnStart())
                 dumpUrl();
         }
-        catch (Exception e)
+        catch (Throwable t)
         {
```

Comment:
Definitely fixes the problem when throwUnavailableOnStartupException==false. However, when it's true, the problem is back again!

Reply:
Yet oddly, `setThrowUnavailableOnStartupException(true);` and `setThrowUnavailableOnStartupException(false);` have the same results.  
The deployment fails and is not tried again by the DeploymentManager as the webapp isn't "started" or "available".

Reply:
So if the webapp is marked as available==false, and the DM doesn't retry it, why is the ContextHandlerCollection startup trying to start it, only iff the UnavailableException is thrown?

Reply:
I added to the tests.

There's a scenario for `setThrowUnavailableOnStartupException(true);` and `setThrowUnavailableOnStartupException(false);` now (declared in XML actually)

The results ...

Context | ThrowOnUnavail...Exception | isStarted | isFailed | isAvailable
-------- | ----------------------------------------- | --------- | ------- | ----------
`/badapp` | true | false | true | false
`/badapp-uaf` | false | true | false | false

In both cases, the webapp is unavailable.
The TrackedConfiguration verifies that the Configuration is only attempted once, and fails half-way through.
The ContextHandlerCollection doesn't try to start it again.
The DeploymentManager doesn't try to start it again.

I could (in the testcase) add an actual GET request against the context's to see if they are truly unavailable (it just feels like that kind of test is done elsewhere already)

Reply:
ContextHandlerCollection startup (doStart()) has no handlers when it executes.

The ContextHandler's are added by the DeploymentManager Bindings to the post-started ContextHandlerCollection.

The Bindings are as follows ...

Binding | Role
-------- | ----
StandardDeployer | ContextHandlerCollection.addHandler(contextHandler)
StandardStarter | ContextHandlerCollection.manage(contextHandler)
StandardStopper | ContextHandlerCollection.unmanage(contextHandler)
StandardUndeployer | ContextHandlerCollection.getHandlers() // remove context recursively

The failures are occurring during StandardStarter.processBinding().